include/xtensor/xsort.hpp

```diff
@@ -697,6 +702,182 @@ namespace xt
+            select_indices_impl(
+                shape, indices, axis, 0, xtl::make_sequence<index_type>(shape.size()), out);
+            return out;
+        }
+
+        // TODO remove when fancy index views are implemented
+        // Poor man's indexing along a single axis as in NumPy a[:, [1, 3, 4]]
+        template<class E, class I>
+        inline auto fancy_indexing(E&& e, I const& indices, std::ptrdiff_t axis)
+        {
+            std::size_t const ax = normalize_axis(e.dimension(), axis);
+            using shape_t = get_strides_t<typename std::decay_t<E>::shape_type>;
+            auto shape = e.shape();
+            shape[ax] = indices.size();
+            return reshape_view(
+                index_view(std::forward<E>(e), select_indices(e.shape(), indices, ax)),
+                std::move(shape)
+            );
+        }
+
+        template<class T, class I, class P>
+        inline auto quantile_kth_gamma(std::size_t n, P const& probas, T alpha, T beta)
+        {
+            auto const m = alpha + probas * (T(1) - alpha - beta);
+            auto const p_n_m = probas * static_cast<T>(n) + m;
+            auto const j = floor(p_n_m);
+            return std::make_pair(
+                /* kth= */ eval(xt::cast<std::size_t>(j) - 1),  // jth index in article is 1-based
+                /* gamma= */ eval(p_n_m - j)
+                );
+        }
+
+        // TODO should implement unsqueeze rather
+        template<class S>
+        inline auto unsqueeze_shape(S const& shape, std::size_t axis)
+        {
+            XTENSOR_ASSERT(axis <= shape.size());
+            auto new_shape = xtl::forward_sequence<xt::svector<std::size_t>, decltype(shape)>(shape);
+            new_shape.insert(new_shape.begin() + axis, 1);
+            return new_shape;
+        }
+
+        template<class S>
+        inline S swapaxes_dims(std::size_t dim, std::ptrdiff_t axis1, std::ptrdiff_t axis2)
+        {
+            std::size_t const ax1 = normalize_axis(dim, axis1);
+            std::size_t const ax2 = normalize_axis(dim, axis2);
+            auto perm = xtl::make_sequence<S>(dim, 0);
+            using id_t = typename S::value_type;
+            std::iota(perm.begin(), perm.end(), id_t(0));
+            perm[ax1] = ax2;
+            perm[ax2] = ax1;
+            return perm;
+        }
+
+        // TODO should implement swapaxis and moveaxis
+        template<class E>
+        inline auto swapaxes(E&& e, std::ptrdiff_t axis1, std::ptrdiff_t axis2)
+        {
+            using strides_t = get_strides_t<typename std::decay_t<E>::shape_type>;
+            return xt::transpose(
+                std::forward<E>(e),
+                swapaxes_dims<strides_t>(e.dimension(), axis1, axis2)
+            );
+        }
+
+    }
+
+        // TODO doc as in article
+        // TODO thanks Julia
+    template<class T = double, class E, class P>
+    inline auto quantile(E&& e, P const& probas, std::ptrdiff_t axis, T alpha = 1., T beta = 1.)
```

Comment:
@JohanMabille Here `proba` needs to be xtensor capable.
What's a good way to do that in a generic way, while allowing `std::array` and `std::vector`? Using `xt::adapt`? or letting the user make a conversion?